packages/labs/src/tag-input/tagInput.tsx

```diff
@@ -167,8 +171,13 @@ export class TagInput extends AbstractComponent<ITagInputProps, ITagInputState>
         const { selectionEnd, value } = event.currentTarget;
         if (event.which === Keys.ENTER && value.length > 0) {
             // enter key on non-empty string invokes onAdd
-            this.setState({ inputValue: "" });
-            Utils.safeInvoke(this.props.onAdd, value);
+            const shouldClearInput = Utils.safeInvoke(this.props.onAdd, value);
+
+            // strict equal is necessary to ensure the input is cleared whether the return value is
+            // true or undefined
```

Comment:
PR title does not match implementation. "Return boolean from onAdd that [does not clear input if false]"